### PR TITLE
Feature/te5 t406 android(b) UI widgets

### PR DESCRIPTION
**Pull request by BrijeshBarasiya**
Linear Layout
Constraint Layout
Relative Layout
Frame Layout
Table Layout
Grid Layout
Tab Layout
Include, Merge Tag

### CODE REVIEW

File: app/src/main/java/com/example/basicofkotlin/RecyclerviewAdapter.kt [hunk 1]
```diff
@@ -31,10 +32,14 @@ class RecyclerviewAdapter(val labelList: List<String>): RecyclerView.Adapter<Rec
             exerciseLabel = view.lblListItem
             exerciseLabel.setOnClickListener{
                 when(exerciseLabel.text) {
-                    "Chapter: 3(A) Ui Widgets" -> {
+                    "Chapter: 3(A) Ui-Widgets" -> {
                         var intent = Intent(view.context, FormExercise:: class.java)
```
Comment (nishchal-v-simformsolutions) [inline app/src/main/java/com/example/basicofkotlin/RecyclerviewAdapter.kt:36]: we can handle this based on the click position and not on the text of the widget, please let me know if you don't understand,
this logic needs to be fixed

File: app/src/main/java/com/example/uiwidgets/LayoutExercise.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        demoTableLayout.setOnClickListener {
+            var intent = Intent(applicationContext, TableLayoutDemo:: class.java)
+            startActivity(intent)
+        }
+
+        demoGridLayout.setOnClickListener {
+            var intent = Intent(applicationContext, GridLayoutDemo:: class.java)
+            startActivity(intent)
+        }
+
```
Comment (nishchal-v-simformsolutions) [inline app/src/main/java/com/example/uiwidgets/LayoutExercise.kt:50]: remove this extra lines before all  setOnClickListener

File: app/src/main/res/layout/activity_java_practical.xml [hunk 3]
```diff
@@ -12,26 +11,26 @@
...
 
     <TextView
         android:id="@+id/lbl_answer"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:textSize="24dp"
+        android:text=""
         android:textColor="@color/black"
-        android:text=""/>
+        android:textSize="24dp" />
```
Comment (nishchal-v-simformsolutions) [inline app/src/main/res/layout/activity_java_practical.xml:34]: textSize is always used in Sp and we always use SSP

File: app/src/main/res/layout/tab_layout_demo.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    tools:context="com.example.uiwidgets.TabLayoutDemo">
+
+    <com.google.android.material.tabs.TabLayout
+        android:id="@+id/tblyDemo"
```
Comment (nishchal-v-simformsolutions) [inline app/src/main/res/layout/tab_layout_demo.xml:10]: always use snake case in XML ids

**Review by nishchal-v-simformsolutions — CHANGES_REQUESTED** [review]
PR size should be around 20 ( you can do separate PR if the limit exceeds ) 
SSP and SDP are not used in place of sp and dp
XML component id naming should be in snake_case

File: app/src/main/res/layout/tab_layout_demo.xml (hunk 4 above)
Comment (BrijeshBarasiya) [reply app/src/main/res/layout/tab_layout_demo.xml:10]: It will Work or i need to change it?

**Pull request merged**